platform/api/oci/instance.go

```diff
@@ -45,98 +50,133 @@ func (a *API) CreateInstance(name, userdata, sshKey string) (*Machine, error) {
 		metadata["ssh_authorized_keys"] = sshKey
 	}
 
-	opts := baremetal.LaunchInstanceOptions{
-		Metadata: metadata,
-		CreateVnicOptions: &baremetal.CreateVnicOptions{
-			AssignPublicIp: boolToPtr(true),
-			SubnetID:       subnet.ID,
-			HostnameLabel:  name,
-		},
-		HostnameLabel: name, // hostname label must match vnic hostname label
-		CreateOptions: baremetal.CreateOptions{
-			DisplayNameOptions: baremetal.DisplayNameOptions{
-				DisplayName: name,
+	resp, err := a.compute.LaunchInstance(context.Background(), core.LaunchInstanceRequest{
+		LaunchInstanceDetails: core.LaunchInstanceDetails{
+			AvailabilityDomain: subnet.AvailabilityDomain,
+			CompartmentId:      &a.opts.CompartmentID,
+			Shape:              &a.opts.Shape,
+			CreateVnicDetails: &core.CreateVnicDetails{
+				SubnetId:       subnet.Id,
+				AssignPublicIp: boolToPtr(true),
+				HostnameLabel:  &name,
 			},
+			DisplayName:   &name,
+			HostnameLabel: &name,
+			ImageId:       &a.opts.Image,
+			Metadata:      metadata,
 		},
-	}
-
-	inst, err := a.client.LaunchInstance(subnet.AvailabilityDomain, a.opts.CompartmentID, a.opts.Image, a.opts.Shape, subnet.ID, &opts)
+	})
 	if err != nil {
 		return nil, err
 	}
 
-	id := inst.ID
+	inst := resp.Instance
+	id := inst.Id
+
+	if id == nil {
+		return nil, fmt.Errorf("received instance id nil")
+	}
 
 	err = util.WaitUntilReady(10*time.Minute, 10*time.Second, func() (bool, error) {
-		inst, err = a.client.GetInstance(id)
+		getInst, err := a.compute.GetInstance(context.Background(), core.GetInstanceRequest{
+			InstanceId: id,
+		})
 		if err != nil {
 			return false, err
 		}
+		inst = getInst.Instance
 
-		if inst.State != "RUNNING" {
+		if inst.LifecycleState != "RUNNING" {
 			return false, nil
 		}
 
 		return true, nil
 	})
 	if err != nil {
-		a.TerminateInstance(id)
+		a.TerminateInstance(*id)
 		return nil, fmt.Errorf("waiting for machine to become active: %v", err)
 	}
 
-	vnicOpts := baremetal.ListVnicAttachmentsOptions{
-		InstanceIDListOptions: baremetal.InstanceIDListOptions{
-			InstanceID: inst.ID,
-		},
+	if inst.Id == nil {
```

Comment:
move this check about the TerminateInstance command.

Reply:
The earlier `TerminateInstance` calls uses the `id` variable which is checked before hand. Inside of the `WaitUntilReady` we are updating the `inst` variable with the output of `GetInstance`, this check is to make sure that the new output also doesn't have a nil `Id`.

Are you wanting me to move this check to after each update of the `inst` variable and swap the `TerminateInstance` call to use `inst.Id`?

Reply:
Oh, woops, got `id` and `inst.Id` mixed up. This is good.